web_src/js/features/repo-common.js

```diff
@@ -1,38 +1,30 @@
 import $ from 'jquery';
 import {hideElem, showElem} from '../utils/dom.js';
+import {POST} from '../modules/fetch.js';
 
-const {csrfToken} = window.config;
+async function getArchive($target, url, first) {
+  const response = await POST(url);
+  if (response.status === 200) {
+    const data = await response.json();
+    if (!data) {
+      // XXX Shouldn't happen?
+      $target.closest('.dropdown').children('i').removeClass('loading');
```

Comment:
put it in a `try-finally` and access  the data within. This is important because fetch will throw on network errors.

Reply:
It shouldn't happen always so I put it in a `catch` but anyway these loading indicators are not working

Reply:
Now you have 3 code branches that remove the indicator, while a single `try-finally` would be sufficient as long as you don't `return` inside the `try`. Actually I think a `finally` may even execute when you `return`.

Reply:
> Now you have 3 code branches that remove the indicator, while a single `finally` would be sufficient as long as you don't `return` inside the `try`. Actually I think a `finally` may even execute when you `return`.

Yes, `finally` executes always, even after `return`

Reply:
In fact it does:

```js
(function () {
  try {
    return console.log('a');
  } finally {
    console.log('b');
  }
})();
```
Logs both "a" and "b".

Reply:
In the case `complete` is false we don't want to remove the loading indicator

Reply:
How about setting a `isComplete = true` variable before the fetch, set it to `false` conditionally and check it during indicator removal?

Reply:
The code as it is right now has the same behavior as before. Logic changes should not go in this PR

Reply:
The code reads like ass but previous one did too, so whatever.

Reply:
Agreed on that front